cli/packages.go

```diff
@@ -0,0 +1,133 @@
+	PackageTypeArchive     = PackageType("archive")
+	PackageTypeMLModel     = PackageType("ml_model")
+	PackageTypeModule      = PackageType("module")
+	PackageTypeSLAMMap     = PackageType("slam_map")
+)
+
+var packageTypes = []string{
+	string(PackageTypeUnspecified), string(PackageTypeArchive), string(PackageTypeMLModel),
+	string(PackageTypeModule), string(PackageTypeSLAMMap),
+}
+
+// PackageExportAction is the corresponding action for 'package export'.
+func PackageExportAction(c *cli.Context) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+
+	return client.packageExportAction(
+		c.String(generalFlagOrgID),
+		c.String(packageFlagName),
+		c.String(packageFlagVersion),
+		c.String(packageFlagType),
+		c.Path(packageFlagDestination),
+	)
+}
+
+func (c *viamClient) packageExportAction(orgID, name, version, packageType, destination string) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+
+	packageID := path.Join(orgID, name)
+
+	var packageTypeProto packagespb.PackageType
+	switch PackageType(packageType) {
+	case PackageTypeUnspecified:
+		packageTypeProto = packagespb.PackageType_PACKAGE_TYPE_UNSPECIFIED
+	case PackageTypeArchive:
+		packageTypeProto = packagespb.PackageType_PACKAGE_TYPE_ARCHIVE
+	case PackageTypeMLModel:
+		packageTypeProto = packagespb.PackageType_PACKAGE_TYPE_ML_MODEL
+	case PackageTypeModule:
+		packageTypeProto = packagespb.PackageType_PACKAGE_TYPE_MODULE
+	case PackageTypeSLAMMap:
+		packageTypeProto = packagespb.PackageType_PACKAGE_TYPE_SLAM_MAP
+	default:
+		return errors.New("invalid package type " + packageType)
+	}
+
+	resp, err := c.packageClient.GetPackage(c.c.Context,
+		&packagespb.GetPackageRequest{
+			Id:         packageID,
+			Version:    version,
+			Type:       &packageTypeProto,
+			IncludeUrl: &boolTrue,
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	return downloadPackageFromURL(c.c.Context, c.authFlow.httpClient, destination, name, version, resp.GetPackage().GetUrl())
+}
+
+func downloadPackageFromURL(ctx context.Context, httpClient *http.Client,
+	destination, name, version, packageURL string,
+) error {
+	packagePath := filepath.Join(destination, version, name+".tar.gz")
+	if err := os.MkdirAll(filepath.Dir(packagePath), 0o700); err != nil {
+		return err
+	}
+	//nolint:gosec
+	packageFile, err := os.Create(packagePath)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, packageURL, nil)
+	if err != nil {
+		return errors.Wrapf(err, serverErrorMessage)
+	}
+	res, err := httpClient.Do(req)
+	if err != nil {
+		return errors.Wrapf(err, serverErrorMessage)
+	}
+	if res.StatusCode != http.StatusOK {
+		return errors.New(serverErrorMessage)
+	}
+	defer func() {
+		utils.UncheckedError(res.Body.Close())
+	}()
+
+	bin, err := io.ReadAll(res.Body)
+	if err != nil {
+		return errors.Wrapf(err, serverErrorMessage)
+	}
+	r := io.NopCloser(bytes.NewReader(bin))
```

Comment:
[q] All packages are zipped; I am not sure if we want to unzip locally. That would be consistent with the behavior for data export. Any suggestions?

Reply:
There is an inconsistency with how we handle package uploads. For modules, the uploaded format could be file, directory, or compressed archive. For ML models, we require individual files and then zip on upload. For ML training scripts, we will require the user to zip the package first. In the backend, everything is compressed (everything in GCS is stored as a compressed archive). 

I don't think there's an easy way to distinguish between what was compressed before upload versus on upload, so leaving this to download as the zipped package for now. I did make [this ticket](https://viam.atlassian.net/browse/DATA-2481) to make the behavior consistent across the package types we own (ML models + training scripts)

Reply:
Ah interesting! Thanks for making that ticket; solving there first SGTM.